cognitedata/reveal
# fix(react-components): optimize styled node collection diff check on re-render

#### Type of change

![Bug](https://img.shields.io/badge/Type-Bug-red) 

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/

## Description :pencil:


Optimize style-difference check. This severely hampered performance on model 
```
{
    modelId: 3282558010084460,
    revisionId: 4932190516335812
}
```
in 3d-test. This is likely due to the large number of mapped equipment.

Essentially, this optimization ensures index sets are not reconstructed unnecessarily, but stored properly from when edges are constructed.

Update: It seems like I (using Firefox on Linux) am the only one having these severe performance problems. I'm adding a video to document what the experience is at my end:


https://github.com/cognitedata/reveal/assets/70905152/581a8eb8-3cf0-4b05-8079-be10e1aec861



## How has this been tested? :mag:



In FDX locally.


## Test instructions :information_source:



Open https://apps.cognite.com/explore/FDX_Boys/fdx-boys/2/Equipment/fdx-boys/Gokhan-2878543855731702-IAA?aiSearch=false&viewMode=3d and notice how clicks on Ivar Aasen (the oil platform) are sort-of slow.

Use this PR in FDX locally, and see how clicks are notably faster.

## Checklist :ballot_box_with_check:
